connections/evm/connection.go

```diff
@@ -22,76 +22,86 @@ import (
 	"github.com/ethereum/go-ethereum/rpc"
 )
 
+type connection interface { //nolint:deadcode,unused
+	Connect() error
+	Close()
+	NetworkId() (*big.Int, error)
+	LatestBlock() (*big.Int, error)
+	EnsureHasBytecode(addr ethcommon.Address) error
+	NewTransactOpts(value, gasLimit, gasPrice *big.Int) (*bind.TransactOpts, uint64, error)
+	WaitForBlock(block *big.Int) error
+}
```

Comment:
Interface should be defined in the code that depends on them not the code that implements them. 

A fundamental property of Go interfaces is that the implementation does not have to specify the interface it is implementing, entirely decoupling the two and allowing for far more dynamic usages of interfaces.